src/gg.regression.unity.bots/Editor/Scripts/CodeGenerators/RGCodeGenerator.cs

```diff
@@ -107,818 +114,872 @@ public static void GenerateRGScripts()
 
         }
 
-        private static List<Scene> GetDirtyScenes()
+        [MenuItem("Regression Games/Agent Builder/Extract Game Context")]
+        private static void ExtractGameContext()
         {
-            var result = new List<Scene>();
-            for (int j = 0; j < SceneManager.sceneCount; j++)
+            _hasExtractProblem = false;
+                
+            // don't need to generate scripts here because we auto do that on any code change
+            try
             {
-                var scene = SceneManager.GetSceneAt(j);
-                if (scene.isDirty)
+                RGDebug.LogInfo($"Extracting Regression Games context...");
+                
+                EditorUtility.DisplayProgressBar("Extracting Regression Games Agent Builder Data",
+                    "Regenerating scripts", 0.1f);
+
+                GenerateRGScripts();
+                
+                // Find RGStateEntity scripts and generate state info from them
+                // Do NOT include the previous state infos.. so we don't have dupes
+                // This gives us a consistent view across both generated and hand written state class entities
+                EditorUtility.DisplayProgressBar("Extracting Regression Games Agent Builder Data",
+                    "Extracting state info from RGStateEntity classes", 0.2f);
+                var statesInfos = CreateStateInfoFromRGStateEntities();
+                
+                // Do the same for generate RGActionRequest scripts
+                var actionInfos = CreateActionInfoFromRGActionRequests();
+
+                if (_hasExtractProblem)
```

Comment:
Not sure if I am blind, but isn't this always false?

Reply:
`_hasExtractProblem` gets set anytime `RecordError` or `RecordWarning` is called.

These are possibly set at multiple points during the code generation.